pkg/cmd/provisioning/ibmcloud/delete_service_id_test.go

```diff
@@ -0,0 +1,213 @@
+
+				return tempDirName
+			},
+		},
+		{
+			// Tests the delete-service-id command when failed to listServiceIDs
+			name: "deleteServiceIDs when failed to listServiceIDs",
+			mockIBMCloudClient: func(mockCtrl *gomock.Controller) *mockibmcloud.MockClient {
+				mockIBMCloudClient := mockibmcloud.NewMockClient(mockCtrl)
+				mockListServiceID(mockIBMCloudClient, "name-namespace1-secret-name", 1, true)
+				return mockIBMCloudClient
+			},
+			setup: func(t *testing.T) string {
+				tempDirName, err := ioutil.TempDir(os.TempDir(), testDirPrefix)
+				require.NoError(t, err, "Failed to create temp directory")
+
+				testCredentialsRequest(t, "firstcredreq", "namespace1", "secret-name-0", tempDirName)
+
+				return tempDirName
+			},
+			wantErr: true,
+		},
+		{
+			// Tests the delete-service-id command when failed to DeleteServiceID
+			name: "deleteServiceIDs when failed to DeleteServiceID",
+			mockIBMCloudClient: func(mockCtrl *gomock.Controller) *mockibmcloud.MockClient {
+				mockIBMCloudClient := mockibmcloud.NewMockClient(mockCtrl)
+				mockListServiceID(mockIBMCloudClient, "name-namespace1-secret-name", 1, false)
+				mockDeleteServiceID(mockIBMCloudClient, 1, true)
+				return mockIBMCloudClient
+			},
+			setup: func(t *testing.T) string {
+				tempDirName, err := ioutil.TempDir(os.TempDir(), testDirPrefix)
+				require.NoError(t, err, "Failed to create temp directory")
+
+				testCredentialsRequest(t, "firstcredreq", "namespace1", "secret-name-0", tempDirName)
+
+				return tempDirName
+			},
+			wantErr: true,
+		},
+		{
+			// Tests the delete-service-id command when no matching service ID found, in this case command will just run
+			// fine without any error
+			name: "deleteServiceIDs when no matching service ID found",
+			mockIBMCloudClient: func(mockCtrl *gomock.Controller) *mockibmcloud.MockClient {
+				mockIBMCloudClient := mockibmcloud.NewMockClient(mockCtrl)
+				mockListServiceID(mockIBMCloudClient, "", 0, false)
+				return mockIBMCloudClient
+			},
+			setup: func(t *testing.T) string {
+				tempDirName, err := ioutil.TempDir(os.TempDir(), testDirPrefix)
+				require.NoError(t, err, "Failed to create temp directory")
+
+				testCredentialsRequest(t, "firstcredreq", "namespace1", "secret-name-0", tempDirName)
+
+				return tempDirName
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockCtrl := gomock.NewController(t)
+			defer mockCtrl.Finish()
+
+			mockIBMCloudClient := tt.mockIBMCloudClient(mockCtrl)
+
+			credReqDir := tt.setup(t)
+			defer os.RemoveAll(credReqDir)
+
+			if err := deleteServiceIDs(mockIBMCloudClient, "1234", "name", credReqDir, tt.force); (err != nil) != tt.wantErr {
+				t.Errorf("deleteServiceIDs() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func mockListServiceIDWithSameName(client *mockibmcloud.MockClient, name string, count int, fail bool) {
+	var err error
+	if fail {
+		err = fmt.Errorf("failed to get ListServiceID")
```

Comment:
Can the mocked error more closely mimic the actual error types returned by IBMCloud, or is this generic error a fair representation of the kind of error response we would get?

Reply:
It won't make a lot of difference because we aren't comparing any errors and failing if any non-nil error returns, having that said, I will reuse some error from the sdk.